src/lang/attr_functor.h

```diff
@@ -0,0 +1,75 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file attr_functor.h
+ * \brief A way to defined arbitrary function signature with dispatch on common attributes.
+ *
+ * Common attributes include:
+ *  - int, float, str constants
+ *  - array of attributes
+ *  - map of attributes
+ */
+#ifndef TVM_LANG_ATTR_FUNCTOR_H_
+#define TVM_LANG_ATTR_FUNCTOR_H_
+
+namespace tvm {
+
+template <typename FType>
+class AttrFunctor;
+
+#define ATTR_FUNCTOR_DISPATCH(OP)                                       \
+  vtable.template set_dispatch<OP>(                                     \
+      [](const NodeRef& n, TSelf* self, Args... args) {                 \
+        return self->Visit_(static_cast<const OP*>(n.node_.get()),      \
+                            std::forward<Args>(args)...);               \
+      });                                                               \
+
+// Comparator to compare equality among attributes
```

Comment:
The comment doesn't seem to apply to the AttrFunctor in general (perhaps we'll have other uses later)